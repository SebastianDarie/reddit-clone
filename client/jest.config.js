module.exports = {
  roots: ['<rootDir>'],
  collectCoverage: true,
  collectCoverageFrom: [
    '**/*.{js,jsx,ts,tsx}',
    '!**/*.d.ts',
    '!**/node_modules/**',
    '!**/.next/**',
    '!**/.vercel/**',
  ],
  setupFilesAfterEnv: ['<rootDir>/setupTests.ts'],
  testEnvironment: 'jsdom',
  testPathIgnorePatterns: [
    '<rootDir>[/\\\\](node_modules|.next)[/\\\\]',
    '<rootDir>[/\\\\]src[/\\\\]__tests__[/\\\\]__mocks__',
  ],
  testRegex: '((\\.|/*.)(test))\\.(ts|js)?(x)?$',
  transform: {
    '^.+\\.(ts|tsx)$': 'babel-jest',
  },
  verbose: true,
};
