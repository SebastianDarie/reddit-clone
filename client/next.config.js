module.exports = {
  future: {
    webpack5: true,
  },
  images: {
    domains: ['dummyimage.com', 'd2cqrrc2420sv.cloudfront.net'],
  },
};
