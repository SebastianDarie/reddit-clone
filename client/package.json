{
  "name": "with-chakra-ui-typescript",
  "version": "1.0.0",
  "scripts": {
    "dev": "next",
    "build": "next build",
    "start": "next start",
    "deploy": "vercel --prod",
    "gen": "graphql-codegen --config codegen.yml",
    "test": "jest",
    "test:update": "jest -u",
    "test:pages": "jest --watch --testPathPattern=client/src/__tests__/pages"
  },
  "dependencies": {
    "@apollo/client": "^3.6.9",
    "@chakra-ui/icons": "^1.1.7",
    "@chakra-ui/react": "^1.8.8",
    "@chakra-ui/theme-tools": "1.3.6",
    "@emotion/react": "11.9.3",
    "@emotion/styled": "11.9.3",
    "formik": "^2.2.9",
    "framer-motion": "^4.1.17",
    "graphql": "^15.8.0",
    "graphql-tag": "^2.12.6",
    "next": "latest",
    "next-apollo": "^5.0.8",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-is": "^17.0.2"
  },
  "devDependencies": {
    "@graphql-codegen/cli": "1.21.8",
    "@graphql-codegen/typescript": "1.23.0",
    "@graphql-codegen/typescript-operations": "1.18.4",
    "@graphql-codegen/typescript-react-apollo": "^2.3.1",
    "@testing-library/dom": "^7.31.2",
    "@testing-library/jest-dom": "^5.16.4",
    "@testing-library/react": "^11.2.7",
    "@types/node": "^14.18.21",
    "@types/react": "^17.0.47",
    "@types/react-dom": "^17.0.17",
    "@types/react-test-renderer": "^17.0.2",
    "@types/yup": "^0.29.14",
    "next-page-tester": "^0.33.0",
    "react-icons": "^4.4.0",
    "react-test-renderer": "^17.0.2",
    "typescript": "^4.7.4",
    "yup": "^0.32.11"
  },
  "license": "MIT",
  "packageManager": "yarn@3.2.1"
}
