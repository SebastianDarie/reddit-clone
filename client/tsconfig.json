{
  "compilerOptions": {
    "allowJs": true,
    "alwaysStrict": true,
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true,
    "isolatedModules": true,
    "jsx": "preserve",
    "lib": ["DOM", "ES2020", "ESNext"],
    "module": "esnext",
    "moduleResolution": "node",
    "noEmit": true,
    "noFallthroughCasesInSwitch": true,
    "noUnusedLocals": true,
    "noImplicitAny": true,
    "noUnusedParameters": true,
    "resolveJsonModule": true,
    "skipLibCheck": true,
    "sourceMap": true,
    "strict": true,
    "target": "esnext"
  },
  "exclude": ["node_modules", "src/__tests__"],
  "include": ["**/*.ts", "**/*.tsx"]
}
