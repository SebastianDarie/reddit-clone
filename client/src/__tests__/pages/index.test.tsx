import { getPage } from 'next-page-tester';
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
//import { cleanup, screen } from '../testUtils';
import Index from '../../pages/index';
import { PostsDocument } from '../../generated/graphql';

describe('Index page', () => {
  it('should match the snapshot', () => {
    const component = renderer.create(
      <MockedProvider addTypename={false}>
        <Index />
      </MockedProvider>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});

describe('use @apollo/client', () => {
  test('render posts', async () => {
    const mocks:
      | readonly MockedResponse<Record<string, unknown>>[]
      | undefined = [
      {
        request: {
          query: PostsDocument,
          variables: {
            limit: 5,
            cursor: null,
          },
        },
        result: {
          data: {
            posts: {
              posts: [
                {
                  id: 237,
                  title: 'new post by tim',
                  text: 'Tim gang',
                  points: 1,
                },
                {
                  id: 236,
                  title: 'new apollo post',
                  text: 'first post with apollo wip',
                  points: 2,
                },
                {
                  id: 235,
                  title: 'jcxkcjb jhbhb hxbjxbchbxcjb',
                  text: 'cbx hxb hj hxb b bjhx bhxb jcb ',
                  points: 0,
                },
                {
                  id: 234,
                  title: 'cjh bjh bx jxchbjhcbjh bch',
                  text: 'bc jkbxcjh bhxc cb',
                  points: 0,
                },
                {
                  id: 233,
                  title: 'jbhbhjbchcsbchbcdhj',
                  text: 'cbsjhcbdhjbschbshcsdjh',
                  points: 0,
                },
              ],
            },
          },
        },
      },
    ];

    const { render } = await getPage({
      route: '/',
      wrapper: {
        Page: (Page) => (pageProps) => {
          return (
            <MockedProvider mocks={mocks}>
              <Page {...pageProps} />
            </MockedProvider>
          );
        },
      },
    });
    render();

    await screen.findByText('new post by tim');
  });
});
