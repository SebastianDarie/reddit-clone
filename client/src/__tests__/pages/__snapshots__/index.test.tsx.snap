// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index page should match the snapshot 1`] = `
Array [
  <div
    className="css-b2pafk"
  >
    <div
      className="css-zdacg6"
    >
      <a
        className="chakra-link css-0"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <h2
          className="chakra-heading css-0"
        >
          NRG Reddit
        </h2>
      </a>
      <div
        className="css-zdpt2t"
      >
        loading please wait
      </div>
    </div>
  </div>,
  <div
    className="css-83x9sm"
  >
    <div>
      loading...
    </div>
  </div>,
]
`;
